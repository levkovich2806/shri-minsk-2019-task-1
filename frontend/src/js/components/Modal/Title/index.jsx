import React from "react";
import styles from "./index.module.scss";

function Title(props) {
  const { title = "", onChange } = props;

  return (
    <div className={styles.title}>
      <span className={styles.title__title}>Название: </span>
      <input
        name="title"
        type="text"
        className={styles.title__input}
        onChange={onChange}
        value={title}
      />
    </div>
  );
}

export default Title;
