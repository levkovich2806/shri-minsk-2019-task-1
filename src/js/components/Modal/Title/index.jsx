import React from "react";
import styles from "./index.module.scss";

function Title(props) {
  return (
    <div className={styles.title}>
      <span className={styles.title__title}>Название: </span>
      <input
        type="text"
        className={styles.title__input}
        onChange={props.onChange}
        required
      />
    </div>
  );
}

export default Title;
