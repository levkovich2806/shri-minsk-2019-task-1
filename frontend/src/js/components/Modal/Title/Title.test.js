import React from 'react';
import { shallow, mount } from 'enzyme';

import Title from './index';

describe('Title', () => {

  it('Should match snapshot Title', () => {
    const wrapper = shallow(<Title title="Название заметки" onChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Title check correct input value", () => {

    const wrapper = mount(<Title title="Название заметки" onChange={() => { }} />);

    const input = wrapper.find("input");
    let value = input.get(0).props.value;
    expect(value).toBe("Название заметки");
  });

});
