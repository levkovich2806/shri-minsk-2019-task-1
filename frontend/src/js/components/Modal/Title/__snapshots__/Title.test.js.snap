// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Should match snapshot Title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    onChange={[Function]}
    title="Название заметки"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="title__title"
        >
          Название: 
        </span>,
        <input
          className="title__input"
          name="title"
          onChange={[Function]}
          type="text"
          value="Название заметки"
        />,
      ],
      "className": "title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Название: ",
          "className": "title__title",
        },
        "ref": null,
        "rendered": "Название: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "title__input",
          "name": "title",
          "onChange": [Function],
          "type": "text",
          "value": "Название заметки",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="title__title"
          >
            Название: 
          </span>,
          <input
            className="title__input"
            name="title"
            onChange={[Function]}
            type="text"
            value="Название заметки"
          />,
        ],
        "className": "title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Название: ",
            "className": "title__title",
          },
          "ref": null,
          "rendered": "Название: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "title__input",
            "name": "title",
            "onChange": [Function],
            "type": "text",
            "value": "Название заметки",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
